pkgs/by-name/gp/gpac/package.nix

```
@@ -5,35 +5,97 @@
   cctools,
   pkg-config,
   zlib,
+  ffmpeg,
+  freetype,
+  libjpeg,
+  libpng,
+  libmad,
+  faad2,
+  libogg,
+  libvorbis,
+  libtheora,
+  a52dec,
+  nghttp2,
+  openjpeg,
+  libcaca,
+  libXv,
+  mesa,
+  mesa_glu,
+  xvidcore,
+  openssl,
+  jack2,
+  alsa-lib,
+  pulseaudio,
+  SDL2,
+  curl,
+
+  withFullDeps ? false,
+  withFfmpeg ? withFullDeps,
+  releaseChannel ? "stable",
 }:
 
 stdenv.mkDerivation rec {
   pname = "gpac";
-  version = "2.4.0";
+  version = if releaseChannel == "nightly" then "2.4-unstable-2025-10-26" else "2.4.0";
 
-  src = fetchFromGitHub {
-    owner = "gpac";
-    repo = "gpac";
-    rev = "v${version}";
-    hash = "sha256-RADDqc5RxNV2EfRTzJP/yz66p0riyn81zvwU3r9xncM=";
-  };
+  src =
+    if releaseChannel == "nightly" then
+      fetchFromGitHub {
+        owner = "gpac";
+        repo = "gpac";
+        rev = "e1a54e81b3befba2b0bffd1d4c1cf50da516c5f3";
+        hash = "sha256-jSMBPuWPmTDCebImdmAcCZl0hEQpJK4QMNGcEXgs3A4=";
+      }
+    else
+      fetchFromGitHub {
+        owner = "gpac";
+        repo = "gpac";
+        rev = "v${version}";
+        hash = "sha256-RADDqc5RxNV2EfRTzJP/yz66p0riyn81zvwU3r9xncM=";
+      };
 
-  # this is the bare minimum configuration, as I'm only interested in MP4Box
-  # For most other functionality, this should probably be extended
   nativeBuildInputs = [
     pkg-config
   ]
   ++ lib.optionals stdenv.hostPlatform.isDarwin [
     cctools
+  ]
+  ++ lib.optionals withFfmpeg [
+    ffmpeg
```

This is part of the changes to use the updater.

```suggestion
    channelToUse.ffmpeg-headless
```